Library/Homebrew/dev-cmd/man.rb

```diff
@@ -18,6 +18,10 @@ def man
     else
       regenerate_man_pages
     end
+
+    git_diff_args = ["-C", HOMEBREW_REPOSITORY, "diff", "--stat"]
+    git_diff_args << "--exit-code" if ARGV.flag? "--exit-code"
+    exit system "git", *git_diff_args, "docs/brew.1.html", "manpages"
```

Comment:
I think you can avoid doing any of this unless `--exit-code` is passed. Also, use `ARGV.include? ` instead of `flag?`. I'd also favour setting `Homebrew.failed` instead of calling `exit`.